qt/python/mantidqt/widgets/codeeditor/scriptcompatibility.py

```diff
@@ -0,0 +1,58 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2017 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantid workbench.
+#
+#
+
+# 3rd party imports
+from qtpy.QtWidgets import QMessageBox
+from qtpy.QtGui import QIcon
+
+# local imports
+from mantid import simpleapi
+
+
+def mantid_algorithm_used(content):
+    for attr in dir(simpleapi):
+        if not attr.startswith('_') and attr == attr.title():
+            if attr in content:
```

Comment:
My thinking was it would be quicker to check if the attribute was an algorithm (i.e. title case and does not start with a '_') and _then_ check if the algorithm is in the script's contents, rather than check if every attribute is in `contents`, even if we're not interested in that attribute.
In particular this would be quicker if `contents` is a long string.

Reply:
As far as I'm aware, `attr in contents` would not be evaluated unless the first two conditions were also True, so you don't have to worry about always executing all three conditions

Reply:
Yep. It's not a huge deal, it's just that as it currently is the `attr in content` being in its own "if" reads like it should be a case to give particular attention to, rather than just being another condition to be met. Does that make sense?

Reply:
Yeah I see, only concern with that is PEP8 line length, but I don't know how strictly you guys follow that.

Reply:
encompass the statements with brackets and split the third condition onto a second line. The jenkins Flake8 build will fail if we go over the line length I think